libraries/stdlib/samples/test/samples/collections/arrays.kt

```diff
@@ -64,6 +64,176 @@ class Arrays {
 
     class Transformations {
 
+        @Sample
+        fun associate() {
+            data class Person(val firstName: String, val lastName: String)
+
+            val computerScientists = arrayOf(Person("Grace", "Hopper"), Person("Edsger", "Dijkstra"), Person("Alan", "Turing"))
+
+            val byLastName = computerScientists.associate { it.lastName to it.firstName }
+
+            assertPrints(byLastName, "{Hopper=Grace, Dijkstra=Edsger, Turing=Alan}")
+        }
+
+        @Sample
+        fun associateArrayOfPrimitives() {
+            fun fib(n: Int): Int {
+                var current = 0
+                var next = 1
+                var sum: Int
+                if (n == 0)
+                    return current
+                for (i in 2..n) {
+                    sum = current + next
+                    current = next
+                    next = sum
+                }
+                return next
+            }
+
+            val numbers = intArrayOf(1, 2, 3, 4, 5)
+
+            val fibonacciNumbers = numbers.associate { it to fib(it) }
+
+            assertPrints(fibonacciNumbers, "{1=1, 2=1, 3=2, 4=3, 5=5}")
+        }
+
+        @Sample
+        fun associateBy() {
+            data class Person(val firstName: String, val lastName: String) {
+                override fun toString(): String = "$firstName $lastName"
+            }
+
+            val computerScientists = arrayOf(Person("Grace", "Hopper"), Person("Edsger", "Dijkstra"), Person("Alan", "Turing"))
+
+            val byLastName = computerScientists.associateBy { it.lastName }
+
+            assertPrints(byLastName.keys, "[Hopper, Dijkstra, Turing]")
+            assertPrints(byLastName.values, "[Grace Hopper, Edsger Dijkstra, Alan Turing]")
```

Comment:
Printing keys and values separately made it harder (for me) to see that a map falls out of the `associateBy`. Also, I would change the selection of scientists a bit to include the edge case of repeated keys in the example:
```
val scientists = arrayOf(Person("Grace", "Hopper"), Person("Daniel", "Bernoulli"), Person("Jacob", "Bernoulli"))
val byLastName = scientists.associateBy { it.lastName }
assertPrints(byLastName, "{Hopper=Grace Hopper, Bernoulli=Jacob Bernoulli}")
```

Reply:
Sounds good, will add.